src/modules/cpu/kernel/fmadd_scalar.hpp

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright (c) 2019 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+
+RppStatus fmadd_scalar_f32_f32_host_tensor(Rpp32f *srcPtr,
+                                           RpptGenericDescPtr srcGenericDescPtr,
+                                           Rpp32f *dstPtr,
+                                           RpptGenericDescPtr dstGenericDescPtr,
+                                           Rpp32f *mulTensor,
+                                           Rpp32f *addTensor,
+                                           RpptROI3DPtr roiGenericPtrSrc,
+                                           RpptRoi3DType roiType,
+                                           RppLayoutParams layoutParams,
+                                           rpp::Handle& handle)
+{
+    RpptROI3D roiDefault = {0, 0, 0, (Rpp32s)srcGenericDescPtr->dims[3], (Rpp32s)srcGenericDescPtr->dims[4], (Rpp32s)srcGenericDescPtr->dims[2]};
```

Comment:
Why is the order hardcoded to 3, 4, 2? Won't it be different for a NCDHW srcGenericDescPtr and a NDHWC srcGenericDescPtr since dims[3] is not necessarily always width.

Reply:
Changed this according to srcGenericDescPtr->layout, I missed this change.